File: src/Framework/Core/DataModels/Matrix.DataModels/Knowledge/Knowledge.cs
namespace Matrix.DataModels.Knowledge;

using Matrix.DataModels.Chunks;
using Matrix.DataModels.DataSources;
using Matrix.DataModels.Models;
using System;

public class Knowledge
{
    public Guid KnowledgeUId { get; set; }
    public required String Type { get; set; }
    public DataSourceCollection DataSourceCollection { get; set; } = null!;
    public Model Model { get; set; } = null!;
    public required String Status { get; set; }
    public DataSource OutputDataSource { get; set; } = null!;
    public Chunk ChunkStrategy { get; set; } = null!;
}

File: src/Framework/Portal/Matrix.Portal/src/datamodels/knowledge.ts
import { BaseModel } from './base.model';
import { DataSourceCollection } from './data-source-collection.model';
import { Model } from './model';
import { DataSource } from './data-source.model';
import { Chunk } from './chunk.model';

/**
 * Knowledge interface representing knowledge management entities
 */
export interface Knowledge extends BaseModel {
    knowledgeUId: string;
    name: string;
    description: string;
    type: string;
    dataSourceCollection: DataSourceCollection;
    model: Model;
    status: string;
    outputDataSource: DataSource;
    chunkStrategy: Chunk;
    version: string;
}

/**
 * Knowledge request interface for knowledge processing
 */
export interface KnowledgeRequest {
    query: string;
    type: string;
}

/**
 * Knowledge response interface for knowledge processing
 */
export interface KnowledgeResponse {
    response: string;
    isComplete: boolean;
    progress: number;
}
